### PR TITLE
Md analysis version update

### PR DESCRIPTION
**Pull request by Bernadette-Mohr**
Update `MDAnalysis` to `2.8.0` to support `Gromacs 2024`. Relax Python dependency because `MDAnalysis 2.8.0` requires `Python >= 3.10`, but `nomad-lab` supports `Python >= 3.9`.

### CODE REVIEW

File: atomisticparsers/gromacs/parser.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def __init__(self):
 
     def get_interactions(self, gromacs_version: str = None):
         interactions = super().get_interactions()
-
+        print(interactions)
```
Comment (ladinesa) [inline atomisticparsers/gromacs/parser.py]: remove

File: atomisticparsers/gromacs/parser.py [hunk 2]
```diff
@@ -1098,13 +1099,14 @@ def parse_method(self):
         sec_method.force_field = sec_force_field
         sec_model = Model()
         sec_force_field.model.append(sec_model)
+        print('parse_method, trying to get n_atoms')
```
Comment (ladinesa) [inline atomisticparsers/gromacs/parser.py]: remove

File: atomisticparsers/gromacs/parser.py [hunk 3]
```diff
@@ -1098,13 +1099,14 @@ def parse_method(self):
...
         try:
             n_atoms = self.traj_parser.get('n_atoms', 0)
         except Exception:
             gro_file = self.get_gromacs_file('gro')
             self.traj_parser.mainfile = gro_file
             n_atoms = self.traj_parser.get('n_atoms', 0)
-
+        print('parse_method, getting atoms_info')
```
Comment (ladinesa) [inline atomisticparsers/gromacs/parser.py]: remove

File: atomisticparsers/gromacs/parser.py [hunk 4]
```diff
@@ -1641,15 +1643,27 @@ def write_to_archive(self):
...
         self._frame_rate = None
 
-        sec_run = Run()
-        self.archive.run.append(sec_run)
-
         header = self.log_parser.get('header', {})
 
+        topology_file = self.get_gromacs_file('tpr')
+
+        # try:
```
Comment (ladinesa) [inline atomisticparsers/gromacs/parser.py]: is this something that we can remove?

File: atomisticparsers/utils/mdanalysis.py [hunk 5]
```diff
@@ -74,7 +77,8 @@ def universe(self):
                     self.mainfile, *self.auxilliary_files, **self.options
                 )
             except Exception as e:
-                self.logger.error('Error creating MDAnalysis universe.', exc_info=e)
+                # self.logger.error('Error creating MDAnalysis universe.', exc_info=e)
```
Comment (ladinesa) [inline atomisticparsers/utils/mdanalysis.py]: remove

File: atomisticparsers/gromacs/parser.py [hunk 6]
```diff
@@ -397,6 +396,7 @@ def get_force_field_parameters(self, gromacs_version: str = None) -> List[Dict]:
             )
             return []
         gromacs_version = gromacs_version.split('.')[0] if gromacs_version else None
+        # ! This warning does not get triggered, why?
```
Comment (JFRudzinski) [inline atomisticparsers/gromacs/parser.py]: Can we try to address this comment before merging? Is the version actually "2024" or it's something like "2024.1".

If it's necessary to leave the comment then at least add some info so it is clear what exactly what context the comment is referrring

**Review by JFRudzinski — APPROVED** [review]
See my comment.

File: atomisticparsers/gromacs/parser.py (hunk 6 above)
Comment (Bernadette-Mohr) [reply atomisticparsers/gromacs/parser.py]: No, this is an old comment. An earlier error prevented the code from reaching this check at some point. I removed the comment now.

**Review by Bernadette-Mohr — COMMENTED** [review]
Removed old comment.

**Review by ladinesa — APPROVED** [review]

**Pull request merged**